### ISSUE TITLE
javax.net.ssl.SSLHandshakeException: Handshake failed

### ISSUE THREAD

**Issue by kamilwlf** [android-async-http/android-async-http#1072]
In postman and browser request works properly

backend is on Amazon and uses TLS,
I tried the following code but it does not help:

```
        KeyStore trustStore = trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
        trustStore.load(null, null);

        MySSLSocketFactory socketFactory = new MySSLSocketFactory(trustStore);
        socketFactory.setHostnameVerifier(MySSLSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER);
        client.setSSLSocketFactory(socketFactory);
```

**Comment by dev01-adroitinfosol** [comment]
Here is the solution for you, This can be applied when you are using server for development purpose.

 public SSLContext getSslContext() {

        TrustManager[] byPassTrustManagers = new TrustManager[] { new X509TrustManager() {
            public X509Certificate[] getAcceptedIssuers() {
                return new X509Certificate[0];
            }

            public void checkClientTrusted(X509Certificate[] chain, String authType) {
            }

            public void checkServerTrusted(X509Certificate[] chain, String authType) {
            }
        } };

		SSLContext sslContext=null;

        try {
            sslContext = SSLContext.getInstance("TLS");
        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
        }
        try {
            sslContext.init(null, byPassTrustManagers, new SecureRandom());
        } catch (KeyManagementException e) {
            e.printStackTrace();
        }
        
        return sslContext;

    }
    
     AsyncHttpClient client = new AsyncHttpClient();
      
     client.setSSLSocketFactory(
				new SSLSocketFactory(getSslContext(),
				SSLSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER));

**Comment by hafiz013** [comment]
new SSLSocketFactory(getSslContext(),
                        SSLSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER) has been depreciated. so how?

**Comment by usama27** [comment]
Im facing the same problem, please help.

**Comment by smarek** [comment]
Thanks, this will be solved in #1029, where I'll concentrace issues about using new underlying HttpClient library and cross-support of SSL/TLS versions vs. Server supported versions
